ui/src/logic/useFileUpload.ts

```diff
@@ -0,0 +1,121 @@
+import { useCallback, ChangeEvent, useEffect, useState } from 'react';
+import _ from 'lodash';
+import { PutObjectCommand } from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { dateToDa, deSig } from '@urbit/api';
+import { useFileStore, useStorage } from '@/state/storage';
+import { Upload } from '@/types/storage';
+import api from '../api';
+
+function useFileUpload() {
+  const { s3, ...storage } = useStorage();
+  const { credentials } = s3;
+  const {
+    setFiles,
+    createClient,
+    setStatus,
+    setFileStatus,
+    setErrorMessage,
+    setFileURL,
+    ...fs
+  } = useFileStore();
+  const [hasCredentials, setHasCredentials] = useState(false);
+
+  useEffect(() => {
+    useStorage.getState().initialize(api);
+  }, []);
+
+  useEffect(() => {
+    const hasCreds =
+      credentials?.accessKeyId &&
+      credentials?.endpoint &&
+      credentials?.secretAccessKey;
+    if (hasCreds) {
+      setHasCredentials(true);
+      const connect = async () => {
+        createClient(credentials);
+      };
+      connect()
+        .then(() => setStatus('success'))
+        .catch((error: unknown) => {
+          console.error(error);
+        });
+    }
+  }, [createClient, setStatus, credentials]);
+
+  const uploadFile = useCallback(
+    async (upload) => {
+      const { file, key } = upload;
+      if (fs.client) {
```

Comment:
I think flipping this check and early returning would help make this a little more readable